scipy/signal/_short_time_fft.py

```diff
@@ -0,0 +1,1659 @@
+"""Short-time Fourier Transform Module.
+
+Implementation Notes (as of 2023-04)
+------------------------------------
```

Comment:
These should go into an issue or as comments instead. I think here these will end up in the public documentation which is not wanted.

Reply:
Since `_short_time_fft.py` is not a module, it does not appear in the public documentation.

Reply:
Right, but that's still not a good place for it and we had cases like this were such comments just stay there and are not being used/updated. Our workflow revolves mostly around issues so I think this should go there.